Firebase/Auth/Source/FIRAuth.m

```diff
@@ -512,9 +512,18 @@ - (void)signInWithEmail:(NSString *)email
                password:(NSString *)password
              completion:(FIRAuthResultCallback)completion {
   dispatch_async(FIRAuthGlobalWorkQueue(), ^{
-    [self signInWithEmail:email
-                 password:password
-                 callback:[self signInFlowAuthResultCallbackByDecoratingCallback:completion]];
+    FIRAuthResultCallback decoratedCallback =
+        [self signInFlowAuthResultCallbackByDecoratingCallback:completion];
+    [self internalSignInAndRetrieveDataWithEmail:email
+                                        password:password
+                                      completion:^(FIRAuthDataResult *_Nullable authResult,
+                                                   NSError *_Nullable error) {
+      if (error) {
```

Comment:
It seems that checking for `error` is unnecessary.  Simply calling `decoratedCallback(authResult.user, error);` should work for both cases.

Reply:
Done.